README.md

```diff
@@ -1,5 +1,102 @@
+          ```
+          POST /api/v1/customers
+          Content-Type: application/json
+   
+          {
+            "name": "John Doe",
+            "email": "john@example.com",
+            "age": 30
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+3. **Delete a customer**
+    - **Request:**
+        - Method: DELETE
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          DELETE /api/v1/customers/123
+          ```
+
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Success message confirming deletion.
+
+4. **Update a customer**
+    - **Request:**
+        - Method: PUT
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          PUT /api/v1/customers/123
+          Content-Type: application/json
+   
+          {
+            "name": "Updated Name",
+            "email": "updated@example.com",
+            "age": 35
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+ These endpoints are defined in the `CustomerController` class.
+
+### How the Test works
+- The  integration test spins up the controller, service and database layer. As part of the test it uses the TestContainers 
+library to set up a Postgres container.
+- The ``CustomerControllerTest.Initializer`` static class overrides the application properties in ``application.yml`` so that our 
+application points to the Postgres container
+- Next liquibase runs the database migrations defined in the `migrations.xml` file to set up the customer table in the PostgreSQL container so that the Postgres container contains a Customer table.
+- After, JDBI connects to our Postgres container to add test data into our Postgres container (e.g. adding and removing customers from our table) for our tests to work. 
+An @AfterEach hook runs after each test to clear away the test data.
+
+###  How the Application works
+- The application uses SpringDoc OpenAPI to generate Swagger documentation. The `CustomerApplication` class is annotated with @OpenAPIDefinition to provide metadata for the API documentation e.g titles etc.
+ In the `CustomerController` class the `@RestController` and `@RequestMapping` annotations define the base URL for the endpoints. The `@Tag `annotation provides a description for the API documentation.
+- The SwaggerUI is configured in the `application.yml` file with path to access it. It can be accessed at http://localhost:9000/swagger-ui.html
+when the application is running.
+The application.yml file contains the app's configuration details, like Liquibase config for database.
+- Liquibase reads and applies database migrations defined in `00001_create_table_customer.sql` to ensure the database schema is up-to-date.
```

Comment:
"to ensure the database schema is up-to-date". No it doesn't. Remember at this point in time there is just a database and no tables or anything. You have to say explicitly it applies the migrations and creates the `Customer` table.